c/driver/sqlite/CMakeLists.txt

```diff
@@ -43,6 +43,10 @@ add_arrow_lib(adbc_driver_sqlite
               utils.c
               OUTPUTS
               ADBC_LIBRARIES
+              CMAKE_PACKAGE_NAME
+              AdbcDriverSqlite
```

Comment:
How about using `AdbcDriverSQLite` because SQLite's CMake package name is `SQLite3` not `Sqlite3`?